business/namespaces.go

```diff
@@ -129,12 +127,93 @@ func (in *NamespaceService) GetNamespaces(ctx context.Context) ([]models.Namespa
 		labelSelectorExcludeValue = excludeLabelList[1]
 	}
 
+	namespaces := []models.Namespace{}
+	// Merge namespaces from different clusters
+	// Namespace sameness
+	nsmap := make(map[string]models.Namespace)
+
+	if clientFactory == nil {
+		var err error
+		clientFactory, err = kubernetes.GetClientFactory()
+		if err != nil {
+			return nil, fmt.Errorf("Error getting client factory")
+		}
+	}
+	for _, cluster := range clientFactory.GetClusterNames() {
+		kialiClient := clientFactory.GetSAClient(cluster)
+		nsList, error := in.GetNamespacesByClient(kialiClient, cluster)
+		if error == nil {
+			for _, clusterNamespace := range nsList {
+				namespacesAcum := nsmap[clusterNamespace.Name]
+				if namespacesAcum.Name != "" {
+					// Merge data
```

Comment:
we should log some information here if the data already exists from another cluster's namespace but is different from the current cluster's namespace.

For example, what happens if cluster A namespace has sidecar injection annotation set to true but cluster B namespace has the same sidecar annotation set to false? The last one "wins" but we'll never know of this difference. We should at least log this condition so we will have SOME indication of a mismatch. It would be important to know if the labels or annotations are different across clusters.

That said, I can also envision annotations and even labels having different values across clusters (think of things like "last-updated"  or "last-applied-configuration" type of annotations. Logging messages for those will get noisy. Perhaps log these at TRACE level?

My only concern is behavior is going to be different because the user didn't annotate and label the namespaces the same across clusters and we have no way of letting the user know that.

Reply:
Maybe we can send a warning in the case the annotations are not the same? It would be an added value

Reply:
Yes, I would say, in general, if the labels / annotations are not the same, then we should "say something" because it would be weird if we are treating the namespaces as "the same" but the user has labeled/annotated them differently.

I'm sure there will be cases where it is expected some labels/annotations will be different (just based on the fact they are on different clusters) but in general, I would think they should be basically the same and if they are not, we should log something about those differences.

Reply:
I agree that labels and annotations may vary because users possibly will label namespaces on different clusters with for example a "zone" label (east, west).

Maybe we can just gather those labels and annotations that are the same, and if in the UI, eventually we let users to zoom into any of these merged namespaces, we should fetch all the labels and annotations of it.

Reply:
I sent a last push. If we are not going to perform any logic in the frontend, maybe we can just concat the values when we have a different value for different clusters. This is an example: 

![image](https://user-images.githubusercontent.com/49480155/221867969-ddbf4957-50f9-45cd-8cfb-327fcb25eb83.png)

In the case of `topology.istio.io/network=[value1, value2]`

Not sure if this is a valid solution, though. 
The other approach would be to change the data structure returned for: 

- value of labels can be an array. Disadvantage: Maybe a bit more difficult to manipulate
- create a map for each cluster containing their annotations and labels. 

The disadvantage of this, it could be a little bit more difficult to manipulate

Reply:
I think I'd prefer a more verbose format that allows the UI more flexibility.  The UI is the presentation layer, and I think the API should probably make it easy to be flexible in how the data is visualized.

Reply:
I was thinking about it and created this new branch: https://github.com/kiali/kiali/pull/5883 As it has some more code changes and I'm not sure if it goes in the right direction. I think it is a good approach, but I would like to have an approval first.

Reply:
I will resolve this comment. The initial approach was changed and each namespace will be unique by name/ cluster